protocol/images/type.go

```diff
@@ -0,0 +1,43 @@
+package images
+
+import (
+	"github.com/status-im/status-go/protocol/protobuf"
+)
+
+func jpeg(buf []byte) bool {
+	return len(buf) > 2 &&
+		buf[0] == 0xFF &&
+		buf[1] == 0xD8 &&
+		buf[2] == 0xFF
+}
+
+func png(buf []byte) bool {
+	return len(buf) > 3 &&
+		buf[0] == 0x89 && buf[1] == 0x50 &&
+		buf[2] == 0x4E && buf[3] == 0x47
+}
+
+func gif(buf []byte) bool {
+	return len(buf) > 2 &&
+		buf[0] == 0x47 && buf[1] == 0x49 && buf[2] == 0x46
+}
+
+func webp(buf []byte) bool {
+	return len(buf) > 11 &&
+		buf[8] == 0x57 && buf[9] == 0x45 &&
+		buf[10] == 0x42 && buf[11] == 0x50
+}
+
+func ImageType(buf []byte) protobuf.ImageMessage_ImageType {
+	if jpeg(buf) {
+		return protobuf.ImageMessage_JPEG
+	} else if png(buf) {
+		return protobuf.ImageMessage_PNG
+	} else if gif(buf) {
+		return protobuf.ImageMessage_GIF
+	} else if webp(buf) {
+		return protobuf.ImageMessage_WEBP
+	}
+
+	return protobuf.ImageMessage_UNKNOWN_IMAGE_TYPE
```

Comment:
Although functionally identical to an if/else chain a `switch` is more idiomatic for implementing a choice from a list of options.

```suggestion
	switch {
	case jpeg(buf):
		return protobuf.ImageMessage_JPEG
	case png(buf):
		return protobuf.ImageMessage_PNG
	case gif(buf):
		return protobuf.ImageMessage_GIF
	case webp(buf):
		return protobuf.ImageMessage_WEBP
	default:
		return protobuf.ImageMessage_UNKNOWN_IMAGE_TYPE
	}
```

Reply:
didn't know you can have function in a case statement!

Reply:
Case statements in Go compile down into if else statements, so you can run anything that results in a bool.